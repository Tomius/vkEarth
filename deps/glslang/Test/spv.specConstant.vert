#version 400

layout(constant_id = 17) const int arraySize = 12;
in vec4 ucol[arraySize];

layout(constant_id = 17) const bool spBool = true;
layout(constant_id = 18) const float spFloat = 3.14;
layout(constant_id = 19) const double spDouble = 3.1415926535897932384626433832795;
layout(constant_id = 22) const uint scale = 2;

out vec4 color;
out int size;

void main()
{
    color = ucol[2];
    size = arraySize;
    if (spBool)
        color *= scale;
    color += float(spDouble / spFloat);
}
