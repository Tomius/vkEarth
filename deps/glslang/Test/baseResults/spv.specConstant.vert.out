spv.specConstant.vert
Warning, version 400 is not yet complete; most version-specific features are present, but some are missing.


Linked vertex stage:


// Module Version 10000
// Generated by (magic number): 80001
// Id's are bound by 39

                              Capability Shader
               1:             ExtInstImport  "GLSL.std.450"
                              MemoryModel Logical GLSL450
                              EntryPoint Vertex 4  "main" 9 14 20
                              Source GLSL 400
                              Name 4  "main"
                              Name 9  "color"
                              Name 14  "ucol"
                              Name 20  "size"
                              Decorate 11 SpecId 17
                              Decorate 22 SpecId 17
                              Decorate 26 SpecId 22
                              Decorate 31 SpecId 19
                              Decorate 32 SpecId 18
               2:             TypeVoid
               3:             TypeFunction 2
               6:             TypeFloat 32
               7:             TypeVector 6(float) 4
               8:             TypePointer Output 7(fvec4)
        9(color):      8(ptr) Variable Output
              10:             TypeInt 32 1
              11:     10(int) SpecConstant 12
              12:             TypeArray 7(fvec4) 11
              13:             TypePointer Input 12
        14(ucol):     13(ptr) Variable Input
              15:     10(int) Constant 2
              16:             TypePointer Input 7(fvec4)
              19:             TypePointer Output 10(int)
        20(size):     19(ptr) Variable Output
              21:             TypeBool
              22:    21(bool) SpecConstantTrue
              25:             TypeInt 32 0
              26:     25(int) SpecConstant 2
              30:             TypeFloat 64
              31:   30(float) SpecConstant 1413754136 1074340347
              32:    6(float) SpecConstant 1078523331
         4(main):           2 Function None 3
               5:             Label
              17:     16(ptr) AccessChain 14(ucol) 15
              18:    7(fvec4) Load 17
                              Store 9(color) 18
                              Store 20(size) 11
                              SelectionMerge 24 None
                              BranchConditional 22 23 24
              23:               Label
              27:    6(float)   ConvertUToF 26
              28:    7(fvec4)   Load 9(color)
              29:    7(fvec4)   VectorTimesScalar 28 27
                                Store 9(color) 29
                                Branch 24
              24:             Label
              33:   30(float) FConvert 32
              34:   30(float) FDiv 31 33
              35:    6(float) FConvert 34
              36:    7(fvec4) Load 9(color)
              37:    7(fvec4) CompositeConstruct 35 35 35 35
              38:    7(fvec4) FAdd 36 37
                              Store 9(color) 38
                              Return
                              FunctionEnd
